use-cases/using-SPDK-vhostuser-and-kata.md

```diff
@@ -0,0 +1,153 @@
+
+## SPDK vhost-user Target Overview
+
+The Storage Performance Development Kit (SPDK) provides a set of tools and
+libraries for writing high performance, scalable, user-mode storage applications.
+
+While, virtio is a efficient way to transport data for virtual environments and
+guests. It is most commonly used in QEMU VMs, where the QEMU itself exposes a
+virtual PCI device and the guest OS communicates with it using a specific Virtio
+PCI driver.
+vhost is a protocol for devices accessible via inter-process communication. It
+uses the same virtio queue layout as Virtio to allow vhost devices to be mapped
+directly to virtio devices. The initial vhost implementation is a part of the
+Linux kernel and uses ioctl interface to communicate with userspace applications.
+vhost-user implements the control plane through Unix domain socket to establish
+virtio queue sharing with a user space process on the same host. So SPDK can expose
+vhost device via vhost-user protocol.
+
+SPDK vhost is a vhost-user slave server. It exposes Unix domain sockets and
+allows external applications to connect. It is capable of exposing virtualized
+storage devices to QEMU instances or other arbitrary processes.
+
+Currently, SPDK vhost-user target can exposes these types of virtualized devices:
+
+- `vhost-user-blk`
+- `vhost-user-scsi`
+- `vhost-user-nvme`
+
+For more information, visit [SPDK](https://spdk.io) and [SPDK vhost-user target](https://spdk.io/doc/vhost.html).
+
+## Install and setup SPDK vhost-user target
+
+### Get source code and build SPDK
+
+Following SPDK [getting started](https://spdk.io/doc/getting_started.html)
+
+### Run SPDK vhost-user target
+
+First, run the SPDK setup.sh script to setup some hugepages for the SPDK vhost
+target application. This will allocate 4096MiB (4GiB) of hugepages, enough for
+the SPDK vhost target and the virtual machine.
+
+```bash
+$ sudo HUGEMEM=4096 scripts/setup.sh
+```
+
+Then, take directory `/var/run/kata-containers/vhost-user` as Kata's vhost-user
+device directory. Make subdirectories for vhost-user sockets and device nodes.
+
+For more details, see section [Host setup for vhost-user devices](#host-setup-for-vhost-user-devices).
+
+```bash
+mkdir -p /var/run/kata-containers/vhost-user/
+mkdir -p /var/run/kata-containers/vhost-user/block/
+mkdir -p /var/run/kata-containers/vhost-user/block/sockets/
+mkdir -p /var/run/kata-containers/vhost-user/block/devices/
+```
+
+Next, start the SPDK vhost target application.  The following command will start
+vhost on CPU cores 0 with all future socket files placed in `/var/run/kata-containers/vhost-user/block/sockets/`.
+
+```bash
+$ sudo app/spdk_tgt/spdk_tgt -S /var/run/kata-containers/vhost-user/block/sockets/ &
+```
+
+To list all available vhost options use the following command.
+
+```bash
+$ app/spdk_tgt/spdk_tgt -h
+```
+
+Create a experimental `vhost-user-blk` device based on memory directly.
```

Comment:
- s/a/an/
- Can you change the period to a colon please (here and everywhere else that you are introducing a command to run).